velox/common/memory/MemoryAllocator.cpp

```diff
@@ -381,6 +387,12 @@ bool MemoryAllocatorImpl::allocateContiguousImpl(
     } catch (std::exception& e) {
       // If the new memory reservation fails, we need to release the memory
       // reservation of the freed contiguous and non-contiguous memory.
+      LOG(ERROR) << "Failed to reserve " << numNeededPages * kPageSize
```

Comment:
Should this be a warning and not error. I assume we expect this to happen in tight memory environments.